### PR TITLE
Plotting updates and minor tweaks

### PR DESCRIPTION
**Pull request by ireed3597**

### CODE REVIEW

File: Background/scripts/control_regions.py [hunk 1]
```diff
@@ -86,6 +106,7 @@ def main(f_in, cat, f_out):
     #dy_norm = ROOT.RooRealVar("bkg"+suffix+"_norm","bkg"+suffix+"_norm",1-bkg_frac.getVal(),0,n_events)
 
     c = ROOT.TCanvas("canvas", "canvas", 600, 600)
+    frame.SetAxisRange(65.0,120)
```
Comment (VourMa) [inline Background/scripts/control_regions.py]: Does the addition of this line mean that we all the plots that we save by default with this script will be restricted in the [65, 120] range?

File: Background/tools/plottingTools.py [hunk 2]
```diff
@@ -116,8 +116,8 @@ def plotPdfMap(model,pdfs,plotBlindingRegion,_outdir='./',_cat='',_pdfNBins=1600
     canv.SaveAs("%s/bkgmodel_pdfs_%s.png"%(_outdir,_cat))
     canv.SaveAs("%s/bkgmodel_pdfs_%s.pdf"%(_outdir,_cat))
 
-  if model.xvar.getMax() > 180:
-    hists['data'].GetXaxis().SetRangeUser(100,180)
+  if model.xvar.getMax() > 120:
+    hists['data'].GetXaxis().SetRangeUser(65,120)
```
Comment (VourMa) [inline Background/tools/plottingTools.py:120]: Why this change? It can be breaking for other scripts, as this is a general-usage script.

File: Datacard/makeDatacardGGTT_new.py [hunk 3]
```diff
@@ -223,7 +223,8 @@ def grabYields(df, args):
     df.loc[idx, "sig_yield"] = norm * sf * row.rate
 
     if (args.procTemplate in row.proc) and (row.year == "2016"): # only need bkg yield in one row (choose 2016)
-      bkg_workspace_file = "../Background/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_multipdf_%s_combined.root"%(args.procTemplate, args.MX, args.MY, row["cat"])
+      bkg_workspace_file = "/home/users/evourlio/CMSSW_10_2_13/src/flashggFinalFit/Background/massDecorSRs_noSys_DY/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_multipdf_%s_combined.root"%(args.procTemplate, args.MX, args.MY, row["cat"])
+      #bkg_workspace_file = "../Background/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_multipdf_%s_combined.root"%(args.procTemplate, args.MX, args.MY, row["cat"])
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py]: Probably revert?

File: Datacard/makeDatacardGGTT_new.py [hunk 4]
```diff
@@ -438,16 +439,22 @@ def main(args):
           continue
         catnum = int(cat.split("cat")[1])
         df_row = df[(df.proc=="data_obs")&(df.cat==cat+"cr")].iloc[0]
-        df_row.current_modelWSFile = bkg_workspace_file = "../Background/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_ws_%s_combined.root"%(args.procTemplate, args.MX, args.MY, df_row["cat"])
+        df_row.current_modelWSFile = bkg_workspace_file = "/home/users/evourlio/CMSSW_10_2_13/src/flashggFinalFit/Background/massDecorSRs_noSys_DY/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_ws_%s_combined.root"%(args.procTemplate, args.MX, args.MY, df_row["cat"])
+        #df_row.current_modelWSFile = bkg_workspace_file = "../Background/outdir_%s_combined_mx%dmy%d/fTest/output/CMS-HGG_ws_%s_combined.root"%(args.procTemplate, args.MX, args.MY, df_row["cat"])
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py]: Probably revert?
If we expect that this is a change that people will frequently have to do in the future, maybe make it an argument?

File: Datacard/makeDatacardGGTT_new.py [hunk 5]
```diff
@@ -438,16 +439,22 @@ def main(args):
...
         cr_yield = getNEvents(df_row.current_modelWSFile, df_row.model)
+        upper_bound = cr_yield*2
 
         if catnum == nCats - 1:
-          f.write("\nABCD_A rateParam %scr dy_merged_hgg %d [0,3000000]"%(cat, cr_yield))
+          f.write("\nABCD_A rateParam %scr dy_merged_hgg %d [0,%d]"%(cat, cr_yield, upper_bound))
+          #f.write("\nABCD_A rateParam %scr dy_merged_hgg %d [0,3000000]"%(cat, cr_yield))
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py]: Delete?

File: Datacard/makeDatacardGGTT_new.py [hunk 6]
```diff
@@ -438,16 +439,22 @@ def main(args):
...
           to_add_to_group.append("ABCD_A")
-          f.write("\nABCD_C rateParam %s dy_merged_hgg 5800 [0,10000]"%cat)
+          #TODO find a way to not hard code 5800
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py:448]: What is `5800`? In case any idea comes up about how to get rd of it.

File: Datacard/makeDatacardGGTT_new.py [hunk 7]
```diff
@@ -438,16 +439,22 @@ def main(args):
...
+          f.write("\nABCD_C rateParam %s dy_merged_hgg 5800 [0,%d]"%(cat, upper_bound))
+          #f.write("\nABCD_C rateParam %s dy_merged_hgg 5800 [0,10000]"%cat)
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py]: Delete?

File: Datacard/makeDatacardGGTT_new.py [hunk 8]
```diff
@@ -438,16 +439,22 @@ def main(args):
...
           to_add_to_group.append("ABCD_C")
         else:
-          f.write("\nABCD_B%d rateParam %scr dy_merged_hgg %d [0,20000]"%(catnum, cat, cr_yield))
+          f.write("\nABCD_B%d rateParam %scr dy_merged_hgg %d [0,%d]"%(catnum, cat, cr_yield, upper_bound))
+          #f.write("\nABCD_B%d rateParam %scr dy_merged_hgg %d [0,20000]"%(catnum, cat, cr_yield))
```
Comment (VourMa) [inline Datacard/makeDatacardGGTT_new.py]: Delete?

File: Plots/makeSplusBModelPlot.py [hunk 9]
```diff
@@ -29,7 +29,7 @@ def get_options():
   parser.add_option("--loadSnapshot", dest="loadSnapshot", default=None, help="Load best-fit snapshot name")
   parser.add_option("--cats", dest="cats", default=None, help="Analysis categories. all = loop over cats and plot sum")
   parser.add_option("--unblind", dest="unblind", default=False, action="store_true", help="Unblind signal region")
-  parser.add_option("--blindingRegion", dest="blindingRegion", default="116,134", help="Region in xvar to blind")
+  parser.add_option("--blindingRegion", dest="blindingRegion", default="120,130", help="Region in xvar to blind")
```
Comment (VourMa) [inline Plots/makeSplusBModelPlot.py]: Why change this? It's an argument in any case.

File: Plots/makeToysHH.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+poi_bf = {}
+for poi in opt.POIs.split(","): poi_bf[poi] = w.var(poi).getVal()
+setParamStr = "--setParameters "
+setParam0Str = "--setParameters "
+for p,v in poi_bf.iteritems(): 
+  setParamStr += "%s=%.3f,"%(p,v)
+  setParam0Str += "%s=0,"%p
+if opt.dropResonantBkg: 
+  setParamStr += "r_singleH=0,"
+  setParam0Str += "r_singleH=0,"
```
Comment (VourMa) [inline Plots/makeToysHH.py]: In our case, shouldn't this be setting `res_bkg_scaler`?
https://github.com/cmstas/flashggFinalFit/blob/622c7ad68acb89bc2409a5ef303150e05f183a0b/get_limit_combine.sh#L34

File: Plots/makeToysHH.py [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+if opt.MX is not None: 
+  setParamStr += "MX=%s,"%opt.MX
+  setParam0Str += "MX=%s,"%opt.MX
```
Comment (VourMa) [inline Plots/makeToysHH.py:50]: In our case, also `MY`?

File: Plots/makeToysHH.py [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+    fsub.write("mv higgsCombine_%g_throw_step*.root toy_%g.root\n"%(itoy,itoy))
+    fsub.write("rm gen_%g.root fit_%g.root\n"%(itoy,itoy))
+    fsub.close()
+
+  # Change permission of all files and set running on batch
+  os.system("chmod 775 ./SplusBModels%s/toys/jobs/sub*.sh"%opt.ext)
+  if not opt.dryRun:
+    subs = glob.glob("./SplusBModels%s/toys/jobs/sub*"%opt.ext)
+    for fsub in subs: os.system("qsub -q hep.q -l h_rt=0:20:0 -l h_vmem=12G %s"%fsub)
+  else: print " --> [DRY-RUN] jobs have not been submitted"
```
Comment (VourMa) [inline Plots/makeToysHH.py]: Do we even care to have this?
...although I have started suspecting that this is a "general" script and you just added it?

File: Plots/make_plots.sh [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+make_SpB(){
+    pushd Plots 
+        python makeSplusBModelPlot.py --inputWSFile /home/users/evourlio/CMSSW_10_2_13/src/flashggFinalFit/Combine/massDecorSRs_noSys_DY/Datacard_ggbbres_mx360my90_ggbbres.root  --cat "SR1" --doBands --ext $tag #--parameterMap r:0 #--parameterMap ".*/ttHHggbb.*:r[1,0,2]"
+        #python makeSplusBModelPlot.py --inputWSFile /home/users/iareed/CMSSW_10_2_13/src/flashggFinalFit/Combine/Datacard_ttHHggXX.root --cat "SR1" --doBands --ext ${tag} #--parameterMap r:0 #--parameterMap ".*/ttHHggbb.*:r[1,0,2]"
+        #python makeSplusBModelPlot.py --inputWSFile /home/users/evourlio/CMSSW_10_2_13/src/flashggFinalFit/Combine/massDecorSRs_noSys_DY/Datacard_ggbbres_mx360my90_ggbbres.root --cat "SR2" --doBands --ext $tag #--parameterMap r:0 #--parameterMap ".*/ttHHggbb.*:r[1,0,2]"
+        #python makeSplusBModelPlot.py --inputWSFile /home/users/iareed/CMSSW_10_2_13/src/flashggFinalFit/Combine/Datacard_ttHHggXX.root --cat "SR2" --doBands --ext ${tag} #--parameterMap r:0 #--parameterMap ".*/ttHHggbb.*:r[1,0,2]"
+
+        cp SplusBModels$tag/*.png /home/users/iareed/public_html/plots_for_Giacommo/
+    popd
```
Comment (VourMa) [inline Plots/make_plots.sh]: Probably clean up a bit (absolute directories, what "SR1" is, "ttHHbbgg", etc.)?

**Review by VourMa — COMMENTED** [review]
Mostly cleaning up comments and I must admit I didn't try the plotting scripts. Perhaps @YonsiG should give them a try?
Also, judging by the title, I was expecting to find the script to make the DY plots. Which repos that going to go into? This one feels like the most fitting.

File: Background/tools/plottingTools.py (hunk 2 above)
Comment (ireed3597) [reply Background/tools/plottingTools.py:120]: This only gets invoked just before saving the zoomed in plots of the regular backgrounds. I was making it consistent with the range I've been using for the ABCD plots.

File: Background/scripts/control_regions.py (hunk 1 above)
Comment (ireed3597) [reply Background/scripts/control_regions.py]: Yes, but considering that we had discussed only using the DY estimation for MY near 90 I think this is ok. Either way, I've added a check to only plot this range for MY 80-100. I've also added slightly better control of the reduced fitting range.

File: Datacard/makeDatacardGGTT_new.py (hunk 3 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py]: yep, I had reverted this but must have missed it when fixing some merging

File: Datacard/makeDatacardGGTT_new.py (hunk 4 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py]: Same as above

File: Datacard/makeDatacardGGTT_new.py (hunk 6 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py:448]: It's the value that typically gets saved in the yields of the C workspace file

File: Datacard/makeDatacardGGTT_new.py (hunk 5 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py]: Done

File: Datacard/makeDatacardGGTT_new.py (hunk 7 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py]: Done

File: Datacard/makeDatacardGGTT_new.py (hunk 8 above)
Comment (ireed3597) [reply Datacard/makeDatacardGGTT_new.py]: Done

File: Plots/makeSplusBModelPlot.py (hunk 9 above)
Comment (ireed3597) [reply Plots/makeSplusBModelPlot.py]: This was a change I had made for the ttHH analysis, which these scripts are pulled from. I've reverted it, though for our purpose the blinding region will have to shift for every MY

File: Plots/makeToysHH.py (hunk 10 above)
Comment (ireed3597) [reply Plots/makeToysHH.py]: Yes, corrected

File: Plots/makeToysHH.py (hunk 11 above)
Comment (ireed3597) [reply Plots/makeToysHH.py:50]: Yes, though I have not used the MX option since I've only used this script for the ttHH. I've added it, but we will need to check how this changes the jobs

File: Plots/makeToysHH.py (hunk 12 above)
Comment (ireed3597) [reply Plots/makeToysHH.py]: Not yet, but if we want to run the S+B plots for every mass point that might be something that needs to be used (need ~500 toys per mass point)

That is correct, all the S+B plotting functions are what I got from Franny and adapted to ttHH. I haven't looked into fixing it for X->YH. I wanted it tracked though so others had easier access to it than going to my repo

**Comment by ireed3597** [comment]
> Mostly cleaning up comments and I must admit I didn't try the plotting scripts. Perhaps @YonsiG should give them a try? Also, judging by the title, I was expecting to find the script to make the DY plots. Which repos that going to go into? This one feels like the most fitting.

@YonsiG has run the plotting script and we are debugging them to get them working as intended for the X->YH. I'm not sure what you mean by the DY plots? They are associated with the CRs.

File: Background/tools/plottingTools.py (hunk 2 above)
Comment (VourMa) [reply Background/tools/plottingTools.py:120]: But doesn't affect other plots, possibly? Like plotting the resonant Higgs peak?

File: Datacard/makeDatacardGGTT_new.py (hunk 6 above)
Comment (VourMa) [reply Datacard/makeDatacardGGTT_new.py:448]: I guess it's floated? If so, what about setting it to half the `upper_bound`?

File: Background/scripts/control_regions.py [hunk 14]
```diff
@@ -21,17 +21,19 @@ def main(f_in, cat, f_out):
...
     xvar.setBins(nBinsOutput/4)
 
-    zoom = False
-    if zoom:
-        xvar.setRange(65, 120)
-        xvar.setBins(55)
+    recude_range = False
+    low_bound = 65
+    high_bound = 120
+    if recude_range:
```
Comment (VourMa) [inline Background/scripts/control_regions.py]: Not sure if the typo is consistent throughout, but maybe make the string meaningful, i.e. `reduce_range`?

File: Background/scripts/control_regions.py [hunk 15]
```diff
@@ -106,7 +109,9 @@ def main(f_in, cat, f_out):
     #dy_norm = ROOT.RooRealVar("bkg"+suffix+"_norm","bkg"+suffix+"_norm",1-bkg_frac.getVal(),0,n_events)
 
     c = ROOT.TCanvas("canvas", "canvas", 600, 600)
-    frame.SetAxisRange(65.0,120)
+
+    if (my>=80) and (my<=100):
+        frame.SetAxisRange(65.0,120)
```
Comment (VourMa) [inline Background/scripts/control_regions.py:114]: Do we want this to be controlled by `low_bound` and `high_bound` or not?

File: Background/tools/plottingTools.py (hunk 2 above)
Comment (ireed3597) [reply Background/tools/plottingTools.py:120]: The ```Background``` directory is used exclusively for the non resonant backgrounds. The resonant backgrounds are handled in the ```Signal``` or the ```SignalModelInterpolation``` directories.

File: Plots/makeToysHH.py (hunk 12 above)
Comment (VourMa) [reply Plots/makeToysHH.py]: Let's leave this in for the time being and we can try removing it when testing the jobs and decide then.

File: Background/scripts/control_regions.py (hunk 15 above)
Comment (ireed3597) [reply Background/scripts/control_regions.py:114]: I'd like to keep them distinct since in principal they control two different things, fitting range and plotting range.

File: Background/scripts/control_regions.py (hunk 14 above)
Comment (ireed3597) [reply Background/scripts/control_regions.py]: Fixed

File: Datacard/makeDatacardGGTT_new.py (hunk 6 above)
Comment (VourMa) [reply Datacard/makeDatacardGGTT_new.py:448]: I understand that some tests will be needed for this one, so let's leave it alone for now.

**Review by VourMa — APPROVED** [review]
Thanks for all the work, @ireed3597! I am merging this, and, if need be, let's follow up on another PR with potential changes after testing the scripts
